### PR TITLE
CD-45216 (master) NoMethodError undefined method type_cast for ActiveRecord ConnectionAdapters AbstractMysqlAdapter Column

### PR DESCRIPTION
**Pull request by gubenyas**
reviewers:
- [x] @johnny-lai 
- [x] @abdollar

### CODE REVIEW

File: lib/activerecord-import/import.rb [hunk 1]
```diff
@@ -305,7 +305,7 @@ def values_sql_for_columns_and_attributes(columns, array_of_attributes)   # :nod
           if !sequence_name.blank? && column.name == primary_key && val.nil?
              connection.next_value_for_sequence(sequence_name)
           else
-            connection.quote(column.type_cast(val), column)
```
Comment (gubenyas) [inline lib/activerecord-import/import.rb]: In Rails 4.2 `type_cast` method(https://github.com/rails/rails/blob/v4.1.13/activerecord/lib/active_record/connection_adapters/column.rb#L91) has been removed from the `Column` class and is placed as a private in `ActiveRecord::Type::Value` base class(https://github.com/rails/rails/blob/v4.2.5/activerecord/lib/active_record/type/value.rb#L92). New `Column` class has access to casting methods via `cast_type` object(https://github.com/rails/rails/blob/v4.2.5/activerecord/lib/active_record/connection_adapters/column.rb#L22) which itself is some descendant of `ActiveRecord::Type::Value`(e.g. `Integer` https://github.com/rails/rails/blob/v4.2.5/activerecord/lib/active_record/type/integer.rb). So few new methods are used now in order to do type casting:  `type_cast_from_database`, `type_cast_from_user`, etc.

**Comment by johnny-lai** [comment]
Please update the gemspec file to require rails 4.2

**Comment by johnny-lai** [comment]
looks good otherwise

**Comment by gubenyas** [comment]
@johnny-lai Done

**Comment by johnny-lai** [comment]
Dependency was a little higher than expected. It looks like that file has been there since Rails 4.2.0
https://github.com/rails/rails/blob/v4.2.0/activerecord/lib/active_record/type/integer.rb
So something like `>= 4.2.0` would have been more "correct".

But this issue is a minor issue. So lgtm.

**Comment by abdollar** [comment]
LGTM

**Pull request merged**